pkg/webhook/controlplane/ensurer.go

```diff
@@ -76,23 +76,44 @@ var (
 			},
 		},
 	}
+	// config mount for auditpolicy-webhook-config that is specified at kube-apiserver commandline
+	auditpolicyWebhookConfigVolumeMount = corev1.VolumeMount{
+		Name:      metal.AuditpolicyWebHookConfigName,
+		MountPath: "/etc/webhook/config",
+		ReadOnly:  true,
+	}
+	auditpolicyWebhookConfigVolume = corev1.Volume{
+		Name: metal.AuditpolicyWebHookConfigName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: metal.AuditpolicyWebHookConfigName},
+			},
+		},
+	}
 )
 
 func ensureVolumeMounts(c *corev1.Container) {
 	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, authnWebhookConfigVolumeMount)
 	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, authnWebhookCertVolumeMount)
+	c.VolumeMounts = extensionswebhook.EnsureVolumeMountWithName(c.VolumeMounts, auditpolicyWebhookConfigVolumeMount)
 }
 
 func ensureVolumes(ps *corev1.PodSpec) {
 	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, authnWebhookConfigVolume)
 	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, authnWebhookCertVolume)
+	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, auditpolicyWebhookConfigVolume)
 }
 
 func ensureKubeAPIServerCommandLineArgs(c *corev1.Container) {
 	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--cloud-provider=", "external")
 
 	// activate AuthN Webhook with mounted Webhook-Config
 	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--authentication-token-webhook-config-file=", "/etc/webhook/config/authn-webhook-config.json")
+
+	// activate auditlog with mounted auditpolicy
+	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-policy-file=", "/etc/webhook/config/auditpolicy.yaml")
+	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--audit-log-path=", "/var/log/audit/kube-apiserver-audit.log")
```

Comment:
I think we have to log into a shared volume (sidecar)

Reply:
I have not yet fully understand everything we want to do but do we really have to write a log file into a container? Can't we just push all the logs to somewhere outside of this cluster via an AuditSink?

Reply:
It would be the preferred solution to push the audit-logs directly